filebeat/harvester/registry.go

```diff
@@ -39,9 +45,7 @@ func (r *Registry) Stop() {
 	close(r.done)
 
 	for _, hv := range r.harvesters {
-		r.wg.Add(1)
 		go func(h Harvester) {
-			r.wg.Done()
 			h.Stop()
 		}(hv)
 	}
```

Comment:
That's a good point. The stopping has now issue (at least in my implementation) is that as long as there is a lock, harvesters can't be removed from the list as they also require a lock.